### PR TITLE
Support for configurable TraceListener

### PR DESCRIPTION
**Pull request by pchasco**
We have a requirement to log to a SQL server. The pull request adds two new optional configuration options for the ProxyConfig element:

- logMethod : Optional. can specify "class" to indicate a class will be used for logging.
- logClassTypeName : The fully-qualified class name of the TraceListener to use.

I can work on Java and PHP loggers later if this would be required for acceptance into the repo. It is important for our organization to only use the code in this repository in our projects; I am concerned any local changes we make would be clobbered later if we ever pulled changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ All three proxies respect the XML configuration properties listed below.
         *  .Net levels are "Error", "Warning", "Info", and "Verbose" in order from fewest to most messages; the default is "Error".
         *  Java levels are "SEVERE", "WARNING", "INFO", "CONFIG", "FINE", "FINER", and "FINEST" in order from fewest to most messages; the default is "SEVERE".
         *  PHP levels are 0 (writes messages and errors to logs), 1 (shows proxy errors and messages in browser console), 2 (combination of levels 0 and 1), and 3 (no logging); the default is 0.
-* Add a new `<serverUrl>` entry for each service that will use the proxy. The proxy.config allows you to use the serverUrl tag to specify one or more ArcGIS Server services that the proxy will forward requests to. The serverUrl tag has the following attributes:
+new `<serverUrl>` entry for each service that will use the proxy. The proxy.config allows you to use the serverUrl tag to specify one or more ArcGIS Server services that the proxy will forward requests to. The serverUrl tag has the following attributes:
```
Comment (bsvensson) [inline README.md]: Was this change on purpose?

File: DotNet/README.md [hunk 2]
```diff
@@ -37,6 +37,64 @@ http://[yourmachine]/DotNet/proxy.ashx?http://services.arcgisonline.com/ArcGIS/r
 ```
 * Security tip: By default, the proxy.config allows any referrer. To lock this down, replace the  ```*``` in the ```allowedReferers``` property with your own application URLs.
 
+## .NET Proxy Additional Configuration Settings
+
+The .NET proxy supports specifying an alternative TraceListener implementation for logging. This enables other logging methods such as logging to a database or enterprise logging. You can specify the alternative TraceListener in the proxy config file.
+
+* **logMethod=["class"|"file"|""]** : When set to "class" the log will look at the `logClassTypeName` attribute for the name of a class to use for logging. When set to "file" or if omitted, then the logger will log to a file if such file name is specified with the `logFile` attribute.
```
Comment (bsvensson) [inline DotNet/README.md]: What's the default logMethod? I'm assuming "file"...

Comment (pchasco) [reply DotNet/README.md]: "file". I'll update the doc.

**Comment by pchasco** [comment]
I would like to alter the .NET implementation so that it no longer uses the TraceListener to log from the proxy. The issue with using TraceListener is that TraceListener will also capture other trace events not related to the proxy, for example all Debug.Print messages.

**Comment by pchasco** [comment]
I've fixed the documentation issues and took some time to clean up the implementation.

**Comment by pchasco** [comment]
Hello -- Have you had a chance to review this PR?
